opendistro-for-elasticsearch/sql
# Add JDBC formatter in new SQL engine

*Issue #, if available:* N/A

*Description of changes:* Add a new JDBC formatter to format new engine response as before. This is intend to avoid impact on user after new SQL engine enabled in future. The main differences between this and default JSON formatter include:

1. `alias` field in schema
2. `status` field is added
3. Data type: convert to legacy type name for compatibility, including ES_TEXT, ES_TEXT_KEYWORD, STRUCT, ARRAY.
4. Error response: follow exactly same format as before. Treat syntax and query engine exception as client error.

**Note that this is only for SQL and will take effect without `format` parameter or with `format=jdbc`** New PPL doesn't need this.

*TODO*:

1. Add `version` field and do version check on client side: simply adding new field causes JDBC driver broken, so need to change both sides together later.
2. For now `total` field is always same as `size`: need to use total number of hits as before

*Testing*:

1. Add new `JdbcFormatIT`
2. Add back legacy IT removed earlier due to response format issue: `AggregationExpressionIT` and `QueryIT`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
